tools/secret-management/Azure.Sdk.Tools.SecretRotation.Stores.AzureDevOps/ServiceConnectionParameterStore.cs

```diff
@@ -83,6 +83,11 @@ public class ServiceConnectionParameterStore : SecretStore
         this.logger.LogDebug("Getting service endpoint details for endpoint '{EndpointId}'", endpointId);
         ServiceEndpoint? endpointDetails = await client.GetServiceEndpointDetailsAsync(this.projectName, endpointId);
 
+        if (endpointDetails == null || endpointDetails.Authorization == null)
```

Comment:
This is the actual bug I planned to fix but the tool wouldn't work locally while debugging because of the graph versioning issue so I fixed those as well.